pkg/gen/nodejs-templates/helm.ts.mustache

```diff
@@ -63,63 +63,76 @@ export namespace v2 {
         ) {
             super("kubernetes:helm.sh/v2:Chart", releaseName, config, opts);
 
-            // Create temporary directories and files to hold chart data and override values.
-            const overrides = tmp.fileSync({ postfix: ".yaml" });
-            const chartDir = tmp.dirSync({ unsafeCleanup: true });
-
-            try {
-                let chart: string;
-                let defaultValues: string;
-                if (isChartOpts(config)) {
-                    // Fetch chart.
-                    fetch(`${config.repo}/${config.chart}`, {
-                        destination: chartDir.name,
-                        version: config.version
-                    });
-                    chart = path.quotePath(nodepath.join(chartDir.name, config.chart));
-                    defaultValues = path.quotePath(
-                        nodepath.join(chartDir.name, config.chart, "values.yaml")
-                    );
-                } else {
-                    chart = path.quotePath(config.path);
-                    defaultValues = path.quotePath(nodepath.join(chart, "values.yaml"));
-                }
-
-                // Write overrides file.
-                const data = JSON.stringify(config.values || {}, undefined, "  ");
-                fs.writeFileSync(overrides.name, data);
-
-                // Does not require Tiller. From the `helm template` documentation:
-                //
-                // >  Render chart templates locally and display the output.
-                // >
-                // > This does not require Tiller. However, any values that would normally be
-                // > looked up or retrieved in-cluster will be faked locally. Additionally, none
-                // > of the server-side testing of chart validity (e.g. whether an API is supported)
-                // > is done.
-                const release = shell.quote([releaseName]);
-                const values = path.quotePath(overrides.name);
-                const namespaceArg = config.namespace
-                    ? `--namespace ${shell.quote([config.namespace])}`
-                    : "";
-                const yamlStream = execSync(
-                    `helm template ${chart} --name ${release} --values ${defaultValues} --values ${values} ${namespaceArg}`
-                ).toString();
-                this.resources = this.parseTemplate(yamlStream, config.transformations);
-            } catch (e) {
-                // Shed stack trace, only emit the error.
-                throw new pulumi.RunError(e.toString());
-            } finally {
-                // Clean up temporary files and directories.
-                chartDir.removeCallback();
-                overrides.removeCallback();
-            }
+			const allConfig = pulumi.output(config);
+			const configDeps = Array.from(<Set<pulumi.Resource>>(<any>allConfig).resources());
+
+			(<any>allConfig).isKnown.then((isKnown: boolean) => {
```

Comment:
Just so I'm super clear: this is only true if all the _values_ are known, or if all the _inputs_ are known? It seems to me we want the former, rather than the latter.

Reply:
This is only true if all of the inputs are known. I believe that this is what we want, as any unknown input will prevent us from generating an accurate preview. On a more concrete level, any unknown input will prevent us from running the `apply` below.

Reply:
Hmm, other than `values`, how can other fields affect the preview?

Reply:
Got it. You're right. Let's do it your way.